test/rpc_test.py

```diff
@@ -339,6 +344,21 @@ def test_async_add(self):
         )
         self.assertEqual(fut.wait(), torch.ones(n, n) * 2)
 
+    @dist_init
+    def test_future_timer(self):
+        fut = rpc.rpc_async(
+            "worker{}".format((self.rank + 1) % self.world_size),
+            timeout_function,
+            args=(3,)
+        )
+        time.sleep(1)
```

Comment:
this is not ideal to do in a test, but I'm not sure how to test that the timer works other than this.